src/calendar/types.ts

```diff
@@ -0,0 +1,44 @@
+import { Dayjs } from "dayjs";
+
+export type View = "Day" | "Month" | "Year";
+
+export type VariantDay = "default" | "otherMonth" | "today";
+
+export type VariantMonth = "default" | "currentMonth" | "selectedMonth";
+
+export type VariantYear =
+    | "default"
+    | "currentYear"
+    | "otherDecade"
+    | "selectedYear";
+
+export interface CalendarProps {
+    disabledDates?: string[] | undefined;
+    onChange?: ((value: string) => void) | undefined;
+    value?: string | undefined; // YYYY-MM-DD
+}
+
+export interface CalendarDayProps extends Pick<CalendarProps, "disabledDates"> {
+    selectedStartDate: string;
+    calendarDate: Dayjs;
```

Comment:
i dont recommend you use a lib's type as the prop types. We cannot assume users to download the dependency. You should use either a string or `Date` object then do your own conversion to a `Dayjs` type instead.

Reply:
Apologies. This is not exposed, hence it is fine. Please ignore